### PR TITLE
fix: [CI-5183]: count bitbucket users correctly

### PR DESCRIPTION
**Pull request by jamie-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36110)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36110)

**Comment by jamie-harness** [comment]
trigger messagecheck

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/utils/WebhookTriggerProcessorUtils.java [hunk 1]
```diff
@@ -140,10 +141,16 @@ private static WebhookBaseAttributes convertPushHookBaseAttributes(PushHook push
   }
 
   private static WebhookGitUser convertUser(User user) {
+    String id;
+    if (isEmpty(user.getLogin())) {
+      id = user.getLogin();
```
Comment (hemanthmantri) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/utils/WebhookTriggerProcessorUtils.java:146]: Should this be negation?
!isEmpty()

File: product/ci/scm/proto/scm.proto [hunk 2]
```diff
@@ -42,12 +42,13 @@ enum Action {
...
-  google.protobuf.Timestamp created = 5;
-  google.protobuf.Timestamp updated = 6;
+  string id = 1;
+  string login = 2;
+  string name = 3;
+  string email = 4;
+  string avatar = 5;
+  google.protobuf.Timestamp created = 6;
+  google.protobuf.Timestamp updated = 7;
 }
```
Comment (hemanthmantri) [inline product/ci/scm/proto/scm.proto:52]: Why are we changing the number of proto fields? This breaks compatibility.

Comment (hemanthmantri) [reply product/ci/scm/proto/scm.proto:52]: New fields should be added at the end of proto.

**Review by hemanthmantri — DISMISSED** [review]

**Comment by hemanthmantri** [comment]
Add `go` label for go builds to run.

**Review by rutvijmehta-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 2 of 2 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36110)* status: :shipit: complete! ✓

**Pull request merged**